broker-daemon/orderbook/index.js

```diff
@@ -71,60 +71,73 @@ class Orderbook {
    * @return {Promise} Resolves when market is being watched (not necessarily when it is synced)
    */
   async watchMarket () {
-    this.logger.debug(`Watching market ${this.marketName}...`)
+    let retries = 0
 
-    const { baseSymbol, counterSymbol } = this
-    const { lastUpdated, sequence } = await this.lastUpdate()
-    const params = { baseSymbol, counterSymbol, lastUpdated, sequence }
+    const watchMarketCall = async () => {
```

Comment:
I am not sure this whole thing needs to be wrapped in an anonymous function. It makes it pretty hard to follow. It seems like the anonymous function is being used solely for the purpose of retry logic (unless there is something I am missing). If this is the case, we can pass the retries in as a parameter to watchMarket in the setTimeout call below. (We can default it to 0). :)

Reply:
Hi @martineehrlich, thanks for the feedback!

I agree, the anonymous function makes it hard to follow. I was trying to follow the style used in `lnd-engine` where `validateEngine` uses an anonymous function to handle retries (https://github.com/sparkswap/lnd-engine/blob/628ad14c3448eb2899cc09d796ee93d3e591ea7c/src/index.js#L79-L82), but my goodness is that much cleaner logic than the anonymous function used in `watchMarket`.

Thanks for the suggested changes. I've made the changes and think the logic is _much_ easier to follow. Please let me know if there are any other changes you'd like to see before merging.